static/src/javascripts/projects/journalism/modules/render-campaign.js

```diff
@@ -14,11 +14,13 @@ const renderCampaign = (calloutNode: HTMLElement, calloutData): void => {
             calloutNode.innerHTML = campaignDiv;
         })
         .then(() => {
-            const cForm = document.querySelector(
+            const cForms = document.querySelectorAll(
                 '.element-campaign .campaign .campaign--snippet__body'
             );
-            if (cForm) {
-                cForm.addEventListener('submit', submitForm);
+            if (cForms.length > 0) {
```

Comment:
Hmm, no this will be problematic. The issue is that this function (`renderCampaign`) is called _for each_ callout in the page, which means that if there are 2 callouts in the page, you will end up adding **4** event listeners. There are two solutions:
1. change `document.querySelectorAll` to `calloutNode.querySelector`: this will make sure that you set the listener only to the form inside that campaign; or
2. move this whole `then` block _after_ the `forEach` in `initCampaign`: this will make sure you attach event listeners only once.

Reply:
ah good spot thank you